tests/Application/app/User.php

```diff
@@ -63,7 +63,7 @@ public function scopeActive(Builder $query): Builder
     /** @phpstan-return BelongsTo<Group, User> */
     public function group(): BelongsTo
     {
-        return $this->belongsTo(Group::class);
+        return $this->belongsTo(Group::class)->withTrashed();
```

Comment:
I feel that you should add a new test, and not modify an existing one. cc @canvural

Reply:
Ups, looks like @canvural just merged this. Can you confirm that modifying an existing test is normal?

Reply:
Actually this not a test case. It's just the dummy application that the tests are using. 

So now I think of it, this actually misses a test case 😅 Thank you @nunomaduro 

@IAmRGroot Can you add a new method to https://github.com/nunomaduro/larastan/blob/master/tests/Features/Models/Relations.php called `testNestedSoftDeletesTrait` maybe, that uses the `group` method on `User` class. And return type should be `BelongsTo<Group, User>` You can check how other methods are written in that file.

Reply:
What a turmoil!

Reply:
I did initially change the test application to trigger a check for the nested SoftDeletes trait.

Maybe in hindsight this was not needed as the existing tests `testWithTrashedWithBelongsToRelation`, `testOnlyTrashedWithBelongsToRelation`, `testWithoutTrashedWithBelongsToRelation` already check the `$user->group()->withTrashed()` function:
https://github.com/nunomaduro/larastan/blob/master/tests/Features/Models/Relations.php#L185